public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts

```diff
@@ -55,10 +57,34 @@ export class DashboardScenePageStateManager extends StateManagerBase<DashboardSc
 
     this.setState({ isLoading: true });
 
-    const rsp = await dashboardLoaderSrv.loadDashboard('db', '', uid);
+    let rsp: DashboardDTO | undefined;
 
-    if (rsp.dashboard) {
+    if (uid === DashboardRoutes.Home) {
+      rsp = await getBackendSrv().get('/api/dashboards/home');
+
+      // TODO
+      // if user specified a custom home dashboard redirect to that
+      // if (rsp?.redirectUri) {
+      //   const newUrl = locationUtil.stripBaseFromUrl(rsp.redirectUri);
+      //   locationService.replace(newUrl);
+      // }
+
+      if (rsp?.meta) {
+        rsp.meta.canSave = false;
+        rsp.meta.canShare = false;
+        rsp.meta.canStar = false;
+      }
+    } else {
+      rsp = await dashboardLoaderSrv.loadDashboard('db', '', uid);
+    }
+
+    if (rsp?.dashboard) {
       const scene = transformSaveModelToScene(rsp);
+
+      if (uid === DashboardRoutes.Home) {
+        scene.isHomeDashboard = true;
+      }
```

Comment:
do we need this flag? can't we just set rsp.meta.canEdit = false (as isHomeDashboard is only used in  canEditDashboard)

Reply:
yes, we definitely can - this was an early approach I took before relizing we have meta in state.